ctapipe/coordinates/camera_frame.py

```diff
@@ -19,35 +19,98 @@
 from .representation import PlanarRepresentation
 
 
+def barrel_distortion(x, y, alpha=0):
+    """Apply a barrel distortion to a set of coordinates.
+
+    Parameters
+    ----------
+    x : array_like
+        Cartesian coordinates along the 'x' axis.
+        Shape: (n_pixels)
+    y : array_like
+        Cartesian coordinates along the 'y' axis.
+        Shape: (n_pixels)
+    alpha : float
+        Correction factor for which negative values mean shrinking.
+
+
+    Returns
+    -------
+    xp : array_like
+        Cartesian coordinates along the 'x' axis corrected for barrel distortion.
+        Shape: (n_pixels)
+    yp : array_like
+        Cartesian coordinates along the 'y' axis corrected for barrel distortion.
+        Shape: (n_pixels)
+
+    M.Peresano, K.Kosack, 2020
+    """
+    r = np.hypot(x, y)  # pixel-wise radii
+    xp = x / (1.0 - alpha * r)
+    yp = y / (1.0 - alpha * r)
+    return xp, yp
+
+
+def apply_coma_correction(pix_x, pix_y, alpha=0):
+    """Apply correction for Coma aberration to a set of pixel coordinates.
+
+    This aberration shifts the focused light further away from the center of
+    the camera. The pixels must be moved back before parametrizing to image.
+    The radial shift is proportional to the distance from the center.
+
+    Parameters
+    ----------
+    x : array_like
+        Cartesian coordinates of the pixels along the 'x' axis.
+        Shape: (n_pixels)
+    y : array_like
+        Cartesian coordinates of the pixels along the 'y' axis.
+        Shape: (n_pixels)
+    alpha : float
+        Correction factor equivalent to (1-f) where f is the ratio between
+        effective and nominal focal length of the telescope
+        (https://www.mpi-hd.mpg.de/hfm/CTA/MC/Prod3/Config/PSF/flen.pdf)
+        In this way alpha > 0 means positive distortion.
+
+    Returns
+    -------
+    x_coma : array_like
+        Cartesian coordinates of the pixels along the 'x' axis corrected for
+        Coma aberration.
+        Shape: (n_pixels)
+    y_coma : array_like
+        Cartesian coordinates of the pixels along the 'y' axis corrected for
+        Coma aberration.
+        Shape: (n_pixels)
+
+    M.Peresano, K.Kosack, 2020
+    """
+    # Take the original pixel-coordinates and correct them
+    x_coma, y_coma = barrel_distortion(pix_x.value, pix_y.value, alpha=alpha) * u.m
```

Comment:
This will be wrong if pix_x or pix_y is not in m. Also, why not just use barrel_distortion with units?

Reply:
Perhaps just define alpha as having units of inverse length, then it should all work in any unit system.

Reply:
Is it something different from meters for some cameras?
I will check again the units, as I wrote in the description I have exactly that problem when running the roundtrip test, perhaps is all connected.

Reply:
It does not matter if we currently have cameras not in meters, this kind of thing is an anti-pattern. Functions should that expect unit input should work regardless which units are passed in if they have the correct dimension.

This function does nothing but strip units, call the other function and add a unit back on.
The whole function should not be necessary!

Reply:
I divided the 2 functions because perhaps one want to use barrel distortion in some other cases then Coma correction.
I am working on the units.

Reply:
But barrel distortion also can do pincussion distortion (if alpha has the other sign). And why not just call the distortion functions in the transform.

Also, please don't intermix optical terms. Coma is this:
https://en.wikipedia.org/wiki/Coma_(optics)

This is not directly related to distortion.

Reply:
* Coma is variation of magnification over the entrance pupil

* Distortion is variation of magnification over the angle of incidence